### DISCUSSION TITLE
Disabling browser back and forward gestures

### DISCUSSION THREAD

**Discussion by viralpickaxe** [xyflow/xyflow#3379] [Q&A] [answered]
Hey there,

Is there a prop or some other configuration to disable the native browser two-finger gesture for navigating back and forth through history? We have pan on scroll enabled and it really breaks the interaction. Looking to see if there's a framework solution before doing something more manual 😅

**Comment by chrtze** [comment]
Hey @viralpickaxe, not sure if I understand it correctly: Do you want to disable panOnScroll for this particular interaction and instead navigate the browser? If so, how would you then be able to scroll?

**Reply by viralpickaxe** [reply]
Ah sorry for not being clear here, In Chrome on macOS when swiping with two fingers to the left or right on a website, the browser navigates back or forward in the history stack. This conflicts with the panOnScroll behavior (which we want), when panning around a flow, if I move my two scrolling fingers to the left or the right the browser interprets this as me trying to navigate back, and takes precedence over the react flow pan interaction.

Ideally, there'd be a way to disable this over the canvas

https://github.com/wbkd/react-flow/assets/3677891/fec543b8-da9e-40e8-b427-47d3cd66c267

**Reply by chrtze** [reply]
Can you reproduce it somehow? I have just tested it on our website and it seems to work:

https://github.com/wbkd/react-flow/assets/3797215/d4872bf4-b5f3-47e4-8cef-75a38216bf89

**Reply by viralpickaxe** [reply, marked as answer]
Hrmm this must be something on our side - will keep digging. Thanks for taking a look

**Reply by cbeavis-frameworks** [reply]
Hi. We're having the same issue, has a solution been found?

**Reply by sniperline047** [reply]
I was able to reproduce this issue. (MacOS M1 with Chrome 126.0.6478.183)

When you haven't used the Back gesture while panning on the tab, panning within the React Flow canvas works fine. However, if you've activated the Back gesture even once on that tab and then attempt to pan within the React Flow canvas, the browser's Back gesture takes precedence. As a result, React Flow's `panOnScroll` functionality fails for left-to-right scrolling, though it continues to work correctly for all other directions.

Here's a video attached for the same:

https://github.com/user-attachments/assets/e6d89ed0-c84a-473f-963c-54265f8e7157

**Reply by Rimantovas** [reply]
@viralpickaxe this is still an issue

**Comment by dkimbel** [comment]
One solution that worked for me was to set `overscroll-behavior-x: none;` on the `body` tag (suggested on [StackOverflow](https://stackoverflow.com/a/56071966)). That still allows horizontal `panOnScroll`, but prevents the default macOS Chrome 'back' behavior.

I don't know whether Figma uses React Flow, but from inspecting their HTML, they also set `overscroll-behavior-x: none` -- it seems to be how they avoid that Chrome behavior, too.

**Reply by maliuta-oleksandr** [reply]
Doesn't work for me

**Reply by maliuta-oleksandr** [reply]
found this https://github.com/micho/jQuery.preventMacBackScroll/blob/master/jquery.preventMacBackScroll.js

**Reply by isaiahb** [reply]
Thank you kind sir 🙏

**Reply by svr93** [reply]
@maliuta-oleksandr `html` tag, not `body` ;)
